peoplehr/client.bal

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/http;
+
+# This connector helps you easily integrate People with other systems and applications, for seamless cross-platform data sharing. The People API 
+# accepts and returns JSON data in the request body, with status indicating the outcome of the operation (sucess/failure).
+#
+# + basicClient - People HTTP Client
+@display {label: "PeopleHR Client", iconPath: "resources/peoplehr.svg"}
+public isolated client class Client {
+    final http:Client basicClient;
+    final readonly & ConnectionConfig config;
+
+    # Gets invoked to initialize the `connector`.
+    # The connector initialization requires setting the API credentials. 
+    # Follow [this guide](https://apidocs.peoplehr.com/#section/Authentication) to obtain token for PeopleHR connector.
+    #
+    # + config - People connection configuration
+    # + httpClientConfig - HTTP configuration
+    # + return - `http:Error` in case of failure to initialize or `null` if successfully initialized 
+    public isolated function init(ConnectionConfig config, http:ClientConfiguration httpClientConfig = {}) returns
+                                error? {
+        self.config = config.cloneReadOnly();
+        self.basicClient = check new (config.baseURL, httpClientConfig);
+    }
+
+    # Checks authentication for given user.
+    #
+    # + payload - Request details on authentication
+    # + return -  Successful response or an error
+    @display{label: "Check Authentication"} 
+    isolated remote function checkAuthentication(@display{label: "Authentication Detail"} AuthenticationRequest payload) 
```

Comment:
what does this method actually do? Is it to check if a given user is authenticated at peopleHR? What is returned in the response?